packages/config/src/handlers/updateFormDraft/handler.ts

```diff
@@ -0,0 +1,186 @@
+    }
+
+    draft.history?.unshift(history)
+    draft.status = questionsDraft.status
+    draft.comment = questionsDraft.comment
+    draft.version =
+      draft.status === DraftStatus.PUBLISHED ||
+      draft.status === DraftStatus.PREVIEW ||
+      draft.status === DraftStatus.FINALISED
+        ? draft.version
+        : draft.version + 1
+    draft.updatedAt = Date.now()
+
+    try {
+      await FormDraft.update({ _id: draft._id }, draft)
+    } catch (err) {
+      logger.error(err)
+      return h
+        .response(`Could not update draft for ${draft.event} event`)
+        .code(400)
+    }
+  } else {
+    //create draft
+    try {
+      const formDraft: IFormDraft = {
+        event: questionsDraft.event,
+        status: questionsDraft.status,
+        comment: questionsDraft.comment,
+        version: 1,
+        createdAt: Date.now(),
+        updatedAt: Date.now()
+      }
+      draft = await FormDraft.create(formDraft)
+    } catch (e) {
+      throw internal(e.message)
+    }
+  }
+
+  //update existing questions
+  if (questionsDraft.questions) {
+    const existingQuestions: IQuestionModel[] = await Question.find().exec()
+
+    questionsDraft.questions.forEach(async (question) => {
+      const existingQuestion = existingQuestions.find(
+        (extQuestion) => extQuestion.fieldId === question.fieldId
+      )
+      if (existingQuestion) {
+        try {
+          existingQuestion.fieldId = question.fieldId
+          existingQuestion.label = question.label
+          existingQuestion.placeholder = question.placeholder
+          existingQuestion.maxLength = question.maxLength
+          existingQuestion.fieldName = question.fieldName
+          existingQuestion.fieldType = question.fieldType
+          existingQuestion.preceedingFieldId = question.preceedingFieldId
+          existingQuestion.required = question.required
+          existingQuestion.enabled = question.enabled
+          existingQuestion.custom = question.custom
+          existingQuestion.initialValue = question.initialValue
+
+          await Question.update({ _id: existingQuestion._id }, existingQuestion)
+        } catch (err) {
+          logger.error(err)
+          return h
+            .response(
+              `Could not update question id: ${existingQuestion.fieldId}`
+            )
+            .code(400)
+        }
+      } else {
+        //create new question
+        try {
+          await Question.create(question)
+        } catch (e) {
+          throw internal(e.message)
+        }
+      }
+      return
+    })
+  }
+
+  //delete questions
+  if (questionsDraft.deleted) {
+    questionsDraft.deleted.forEach(async (fieldId) => {
+      try {
+        await Question.findOneAndRemove({ fieldId: fieldId })
+      } catch (err) {
+        return h
+          .response(`Could not delete question fieldId: ${fieldId}`)
+          .code(400)
+      }
+      return
+    })
+  }
+
+  return h.response(draft).code(201)
```

Comment:
This will always return code 201 because the function execution will not wait for the async functions provided in the `forEach`. Actually we should not use async in `forEach` at all

Reply:
Yes. Used map and promise.all instead.